payjoin/src/send/mod.rs

```diff
@@ -255,27 +262,31 @@ impl PsbtContext {
                 (Some((_original_output_index, original_output)), _)
                     if original_output.script_pubkey == self.payee =>
                 {
-                    ensure!(
-                        self.output_substitution == OutputSubstitution::Enabled
-                            || (proposed_txout.script_pubkey == original_output.script_pubkey
-                                && proposed_txout.value >= original_output.value),
-                        DisallowedOutputSubstitution
-                    );
+                    if self.output_substitution == OutputSubstitution::Disabled
+                        || (proposed_txout.script_pubkey == original_output.script_pubkey
+                            && proposed_txout.value < original_output.value)
+                    {
+                        return Err(InternalProposalError::DisallowedOutputSubstitution);
+                    };
```

Comment:
I think this is more strict than the original, it errors if output substitution is disabled, regardless of the output

Reply:
Good catch, I think I have the logic cleaned up here now.